src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -407,6 +413,105 @@ public Response updateDraftVersion( String jsonBody, @PathParam("id") String id,
         }
     }
     
+    @PUT
+    @Path("{id}/addData")
+    public Response addDataToVersion(String jsonBody, @PathParam("id") String id) throws WrappedResponse{
+
+        DataverseRequest req = createDataverseRequest(findUserOrDie());
+
+        return processDatasetUpdate(jsonBody, id, req, true);
+    }
+    
+    @PUT
+    @Path("{id}/updateData")
+    public Response updateDataInVersion(String jsonBody, @PathParam("id") String id)  throws WrappedResponse {
+        
+        DataverseRequest req = createDataverseRequest(findUserOrDie());
+
+        return processDatasetUpdate(jsonBody, id, req, true);
+        
+    } 
+    
+    private Response processDatasetUpdate(String jsonBody, String id, DataverseRequest req, Boolean replaceData){
+        try (StringReader rdr = new StringReader(jsonBody)) {
+           
+            Dataset ds = findDatasetOrDie(id);
+            JsonObject json = Json.createReader(rdr).readObject();
+            DatasetVersion dsv = ds.getEditVersion();
+
+            List<DatasetField> fields = new LinkedList<>();
+            DatasetField singleField = null; 
+            
+            JsonArray fieldsJson = json.getJsonArray("fields");
+            if( fieldsJson == null ){
+                singleField  = jsonParser().parseField(json, Boolean.FALSE);
+                fields.add(singleField);
+            } else{
+                fields = jsonParser().parseMultipleFields(json);
+            }
+
+            dsv.setVersionState(DatasetVersion.VersionState.DRAFT);
+                
+
+            for (DatasetField dsf : dsv.getDatasetFields()) {
```

Comment:
A comment about this parser-loop would probably help future maintenance.

Reply:
Added a comment and also changed the name of the looping variable to add clarity